### PR TITLE
test/old: minor fixes for local invocation

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

File: src/nvim/testdir/Makefile [hunk 1]
```diff
@@ -7,12 +7,11 @@ ifeq ($(OS),Windows_NT)
 else
   NVIM_PRG ?= ../../../build/bin/nvim
 endif
-TMPDIR ?= Xtest-tmpdir
 SCRIPTSOURCE := ../../../runtime
 
 export SHELL := sh
 export NVIM_PRG := $(NVIM_PRG)
-export TMPDIR
+export TMPDIR := Xtest-tmpdir
```
Comment (jamessan) [inline src/nvim/testdir/Makefile]: The changes to this file don't do anything other than make it harder to override `TMPDIR`.  What was "wrong" with the old code?

Comment (justinmk) [reply src/nvim/testdir/Makefile]: On macos TMPDIR is already set to something insane and it makes tests more brittle.

Comment (justinmk) [reply src/nvim/testdir/Makefile]: Note that our Lua tests don't allow TMPDIR override and no one complained.

**Comment by justinmk** [comment]
So regarding configurable $TMPDIR, if we want to make that overridable, we should do that for both `cmake/RunTests.cmake` and `src/nvim/testdir/Makefile`. In the meantime, this avoids spurious failures.

**Pull request merged**